### DISCUSSION TITLE
Metadata and GraphQL

### DISCUSSION THREAD

**Discussion by RStankov** [vercel/next.js#62312] [App Router] [answered]
Hey,

The project I'm working on uses `GraphQL` for data fetching. The team is in the process of migrating from `pages` to `app`.

Currently, data is generated on the server for all our records meta and passed via GraphQL. Here is an example:

```graphql
query PostPage($id: ID) {
  post(id: $id) {
     metadata {
       title
       description
       image
       author
     }
     ...PostPageContentFragment
  }
}
```

My team wants to migrate to the new [Metadata](https://nextjs.org/docs/app/building-your-application/optimizing/metadata) API. 

**However, how can we have a single query in the page component and access the data in `generateMetadata`?**

*I can do 2 queries, but this seems too wasteful. The other option is to ignore the new Metadata API and use something like `Helm`.*

**Comment by markoleavy** [comment, marked as answer]
Doc says just to do two identical queries. React will cache them. Howful, but works. Not sure helm will work in server components.

**Reply by RStankov** [reply]
@markoleavy  Thanks.
